qiime/sdk/artifact.py

```diff
@@ -185,8 +202,16 @@ def _parse_uuid(self, string):
         return uuid.UUID(hex=string)
 
     # TODO implement provenance parsing
-    def _parse_provenance(self, string):
-        return string
+    def _parse_provenance(self, provenance):
+        if isinstance(provenance, str):
```

Comment:
Cool cool, thanks!

On Tue, Apr 12, 2016 at 10:48 AM, Jai Ram Rideout <notifications@github.com>
wrote:

> In qiime/sdk/artifact.py
> <https://github.com/biocore/qiime2/pull/42#discussion_r59420926>:
>
> > @@ -185,8 +202,16 @@ def _parse_uuid(self, string):
> >          return uuid.UUID(hex=string)
> >
> >      # TODO implement provenance parsing
> > -    def _parse_provenance(self, string):
> > -        return string
> > +    def _parse_provenance(self, provenance):
> > +        if isinstance(provenance, str):
>
> This is handling the case where there's no provenance in the artifact.
> Currently if there is no provenance, the serialized artifact provenance is
> the string 'No provenance available because this artifact was generated
> independently of QIIME.'. We're still working out provenance and
> importing data into an artifact so this is just a placeholder for now (#26
> <https://github.com/biocore/qiime2/issues/26>, #31
> <https://github.com/biocore/qiime2/issues/31>).
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiime2/pull/42/files/e72bbd7507985b0633eb3947762645bcca885bb8#r59420926>
>